packit/config/package_config_validator.py

```diff
@@ -95,6 +96,7 @@ def validate(self) -> str:
                     )
                 ) and not self.offline:
                     package_name = package_config.downstream_package_name
+                    self.check_downstream_package_exists(package_name)
```

Comment:
I would say we want to check this in general, not only when pull_from_upstream is defined